### PR TITLE
fixed bug on jss-plugin-nested

### PR DESCRIPTION
**Pull request by heyymarco**
According a bug on [here](https://github.com/cssinjs/jss/issues/1561)

The bug is here:
```
  function replaceParentRefs(nestedProp, parentProp) {
    const parentSelectors = parentProp.split(separatorRegExp)
    const nestedSelectors = nestedProp.split(separatorRegExp)
```
it split the string by comma, but imagine a selector like this:
`&.hello, &.world:is(.awesome, .good, .bad)`

it will splitted to:
```js
['&.hello', '&.world:is(.awesome', '.good', '.bad)']
```
it's incorrect, and should be:
```
['&.hello', '&.world:is(.awesome, .good, .bad)']
```

and the another bug is with the parent selector `&`.  
Imagine if a selector like this:
`&.hello, &.world[data-message="you & me"]`
The `&` inside `"you & me"` is not a parent selector, it's just a string expression.

To solve the 2 problem above, i wrote a **selector parser**,  
you can see (and maybe make more perfection) here: [see my repo](https://github.com/cssfn/css-selector).
The parser is smart enough to distinguish between parent selector, universal selector, combinators, string expression, and so on.
The usage is simple: `parseSelector` => `flatMapSelectors` (replace `&` with something) => `selectorsToString`.

the package published with es9 format, but you can fork & re-transpile to es5.

### CODE REVIEW

**Comment by seiyab** [comment]
Great.
I think you should write tests.

See also:
https://github.com/cssinjs/jss/pull/1568#issuecomment-949967416
https://github.com/cssinjs/jss/pull/1523#issuecomment-869145245
https://github.com/cssinjs/jss/pull/1388#issuecomment-707008924

File: packages/jss-plugin-nested/src/index.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
 import warning from 'tiny-warning'
 
-const separatorRegExp = /\s*,\s*/g
-const parentRegExp = /&/g
+// cssfn:
+import {
+  parseSelectors,
+  flatMapSelectors,
+  selectorsToString,
+}                           from '@cssfn/css-selector'
```
Comment (kof) [inline packages/jss-plugin-nested/src/index.js:8]: I am afraid we can't do this, its a huge bundle-size increase

Comment (kof) [reply packages/jss-plugin-nested/src/index.js:8]: We need a solution that is super minimalistic

Comment (kof) [reply packages/jss-plugin-nested/src/index.js:8]: Wow did you write the selector parser to fix this bug? If so huge props in any case!
But jss is runtime-executed, it gets bundled with a client-side app, the only way forward is to reduce the size, even potentially remove features to make it small.

Comment (haxxxton) [reply packages/jss-plugin-nested/src/index.js:8]: @heyymarco wow dude! thats some serious work!

**Comment by kof** [comment]
please reopen once you have time to continue

**Comment by TchernyavskyDaniil** [comment]
This bug is from version 10.0.1 (more one year!!). Shouldn't it be fixed by the current library team? Part of my team is giving up on jss, as well as some people I know because of this problem. I honestly don't quite understand the point of preparing a move to 18 React if there are such critical bugs
Sorry if I offended or don't understand your processes, I'm speaking as a regular consumer of your library

**Comment by kof** [comment]
Feel free to fix a bug with a proper test and I will merge it.

Oleg

**Pull request closed without merging**